pkg/asset/installconfig/alibabacloud/client.go

```diff
@@ -236,14 +236,58 @@ func (client *Client) ListDNSDomain(baseDomain string) (response *alidns.Describ
 func (client *Client) ListPrivateZones(zoneName string) (response *pvtz.DescribeZonesResponse, err error) {
 	request := pvtz.CreateDescribeZonesRequest()
 	request.Lang = "en"
-	request.Keyword = zoneName
+	if zoneName != "" {
```

Comment:
> I don't understand why this change is necessary. There is no change to request.Keyword when zoneName is an empty string, since request.Keyword is already an empty string.
> Moreover, it seems like a violation of the contract of the ListPrivateZones function to pass in an empty value for zoneName anyway.

I understand your doubts. In the initial assumption, the zoneName of ListPrivateZones is an optional parameter. At present, if users do not want to use zoneName, they must specify an empty string, which is obviously unreasonable.

Reply:
@staebler In the latest update, I used Params type to store parameters

Reply:
@staebler In the latest update, I used Params type to store parameters